tutorials/iot/raspberrypi-wildlife-camera/raspberrypi-wildlife-camera.md

```diff
@@ -0,0 +1,591 @@
+```
+
+Some explanations:
+
+* The `dump` plugin is used to copy files inside the snap, without any build step.
+* The `prime` keyword declares the list of files that will be copied from the `source` ("." indicates the root of the project) into the snap.
+
+The two files we are adding to our snap with this part are:
+
+* `motion-wrapper.sh`: a very small launch wrapper for Motion, where we are going to customise how motion is launched
+* `motion-wildlife-camera.conf`: a Motion configuration file, tuned to take small videos when a certain amount of motion is detected
+
+You can download them by running, from the root of your project:
+
+``` bash
+wget https://raw.githubusercontent.com/caldav/wildlife-camera-tutorial/master/motion-wrapper.sh
+wget https://raw.githubusercontent.com/caldav/wildlife-camera-tutorial/master/motion-wildlife-camera.conf
+```
+
+These two files are heavily commented to explain the purpose of each line, feel free to dive into them!
+
+### Starting Motion as a service
+
+Finally, we want the app to automatically run when the Raspberry Pi starts, to do so we are going to declare it as a service.
+
+To your `snapcraft.yaml`, add the following "apps" entry:
+
+``` yaml
+apps:
+  motion: 
+    command: motion-wrapper.sh
+    daemon: simple
+    plugs:
+      - opengl
+      - camera
+```
+
+We are declaring many important things here. This is how we are turning our snap from a fancy zip file into an actual app that can be launched:
+
+* `apps`: the top level YAML subsection in which apps contained in the snap are declared
+* `motion`: the name of the app, which we are calling "motion" since it's what its going to launch
+* `command`: what the `motion` command will run, meaning our launch wrapper containing our customisation
+* `daemon`: we declare the app as a "simple" daemon, which means it will start as a single process when the host system is turned on
+* `plugs`: a list of interfaces the app can request access to on the host system. Snaps are confined but can still access hardware and host system functionalities through pre-defined interfaces: in this case we will request access to OpenGL system calls and plugged-in video devices.
+
+### Configuration at run time
+
+Before we build our final snap, we are going to add some flair to it. Snaps installed on a system are read-only, and can't be tampered with, but in our case, we will definitely want to be able to test and adjust motion detection to specific framerates, luminosity, or even just change the size of recorded videos.
+
+To do so, we are going to use a "configure hook", a script inside the snap that we will be able to call from the command-line to edit values inside the config file.
+
+Create a `hooks` sub directory in the `snap` directory and enter it:
+
+``` bash
+mkdir snap/hooks
+cd snap/hooks
+```
+
+Now, download the `configure` script:
+
+``` bash
+wget https://raw.githubusercontent.com/caldav/wildlife-camera-tutorial/master/snap/hooks/configure
+```
+
```

Comment:
Cheaters like me would love to wget the original yaml to, to make sure we created it correctly.